internal/putil/putil_test.go

```diff
@@ -171,3 +172,35 @@ func TestParseProviderReference(t *testing.T) {
 		}
 	})
 }
+
+func TestWalk(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		v    property.Value
+		want []r.URN
+	}
+	testCases := []testCase{
+		{
+			v:    property.New("s").WithDependencies([]r.URN{"s"}),
+			want: []r.URN{"s"},
+		},
+		{
+			v: property.New(map[string]property.Value{
+				"k": property.New("s").WithDependencies([]r.URN{"s"}),
+			}).WithDependencies([]r.URN{"k"}),
+			want: []r.URN{"k", "s"},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.v.GoString(), func(t *testing.T) {
+			t.Parallel()
+			var got []r.URN
+			putil.Walk(tc.v, func(v property.Value) (continueWalking bool) {
```

Comment:
If `putil.Walk` is going to return something, shouldn't we test what it returns?